test/unit/BPool/BPool_Bind_Unbind.tree

```diff
@@ -0,0 +1,51 @@
+BPool::bind
+├── when reentrancy lock is NOT set
+│   ├── when called NOT by controller
+│   │   └── it should revert
+│   └── when called by controller
+│       ├── when pool is finalized
+│       │   └── it should revert
+│       ├── when token is already bound
+│       │   └── it should revert
+│       ├── when MAX_BOUND_TOKENS tokens are already bound
+│       │   └── it should revert
+│       ├── when token weight is too low
+│       │   └── it should revert
+│       ├── when token weight is too high
+│       │   └── it should revert
+│       ├── when too little balance is provided
+│       │   └── it should revert
+│       ├── when weight sum exceeds MAX_TOTAL_WEIGHT
+│       │   └── it should revert
+│       └── when it is NOT finalized
```

Comment:
weird, i'd say sth like "when conditions are fulfilled", but "when it is NOT finalized" is wrong imo